### PR TITLE
Overhaul logging

### LINKED ISSUE #313: Chaotic logging
Error messages often get lost in unnecessary noise.
We should use the `logging` module with appropriate log levels.

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #313

### CODE REVIEW

**Comment by sushain97** [comment]
I think this is pretty useful. Maybe it's worth adding proper logging with `logging` and a verbose option to enable this debug output.

**Comment by Androbin** [comment]
Yeah, this is much better.

**Comment by sushain97** [comment]
This looks good to me. Screenshots of how it looks when you start with -v and without?

also @jonorthwash should know about this

**Comment by Androbin** [comment]
without
with

**Comment by sushain97** [comment]
Looks pretty good. How about making it say "Bound module ..." instead of just spitting out those variables. Also, if you could put the registered modules as INFO and ehnologue stuff as DEBUG, might be nice? Basically, clean up the other prints while you're at it. I think the default level should be INFO so that registered modules gets printed.

**Comment by Androbin** [comment]
Okay, I replaced all `print` statements with call to `logger` and the appropriate log level.

File: __init__.py [hunk 1]
```diff
@@ -44,7 +51,7 @@ def connect(config):
 
     try: Watcher()
     except Exception as e:
-        print('Warning:', e, '(in __init__.py)', file=sys.stderr)
+        logger.warning('Warning: ' + str(e) + ' (in __init__.py)')
```
Comment (sushain97) [inline __init__.py]: No need for the double warning here

File: __init__.py [hunk 2]
```diff
@@ -54,8 +61,7 @@ def connect(config):
         if not isinstance(delay, int):
             break
 
-        warning = 'Warning: Disconnected. Reconnecting in %s seconds...' % delay
-        print(warning, file=sys.stderr)
+        logger.warning('Warning: Disconnected. Reconnecting in %s seconds...' % delay)
```
Comment (sushain97) [inline __init__.py]: and here

File: bot.py [hunk 3]
```diff
@@ -70,96 +73,107 @@ def setup(self):
                 module_loader = importlib.machinery.SourceFileLoader(name, filename)
                 module = module_loader.load_module()
             except Exception as e: 
-                print("Error loading %s: %s (in bot.py)" % (name, e), file=sys.stderr)
+                logger.error("Error loading %s: %s (in bot.py)" % (name, e))
             else: 
                 if hasattr(module, 'setup'): 
                     module.setup(self)
-                self.register(vars(module))
+                self.register(module)
```
Comment (sushain97) [inline bot.py]: Seems to be an unrelated change?

File: bot.py [hunk 4]
```diff
@@ -70,96 +73,107 @@ def setup(self):
...
+            if hasattr(func, 'example'):
+                example = func.example
+                example = example.replace('$nickname', self.nick)
+            else: example = None
+
+            self.doc[func.name] = (func.__doc__, example)
+
+        self.commands[priority].setdefault(regexp, []).append(func)
 
-        def bind(self, priority, regexp, func):
```
Comment (sushain97) [inline bot.py]: These are massive changesets, please don't include them in this PR because it makes reviewing much more difficult.

File: modules/git.py [hunk 5]
```diff
@@ -357,7 +360,7 @@ def do_POST_unsafe(self, data):
         self.send_response(200)
         self.send_header("Content-type", "text/html")
         self.end_headers()
-        print("DONE!")
+        logger.debug("DONE!")
```
Comment (sushain97) [inline modules/git.py]: this can probably just go...

File: modules/iso639.py [hunk 6]
```diff
@@ -188,7 +190,7 @@ def setup(phenny):
         try:
             phenny.iso_data = read_dict(f)
         except ValueError:
-            print('iso database read failed, refreshing it')
+            logger.debug('iso database read failed, refreshing it')
```
Comment (sushain97) [inline modules/iso639.py]: these two should be warnings

File: phenny [hunk 7]
```diff
@@ -99,20 +99,21 @@ def create_default_config(fn):
...
+    logger.info('Creating a config directory at ~/.phenny...')
+
     try: os.mkdir(dotdir)
     except Exception as e: 
-        print('There was a problem creating %s:' % dotdir, file=sys.stderr)
-        print(e.__class__, str(e), file=sys.stderr)
-        print('Please fix this and then run phenny again.', file=sys.stderr)
+        logger.error('There was a problem creating %s:' % dotdir)
+        logger.error(e.__class__ + ' ' + str(e))
+        logger.error('Please fix this and then run phenny again.')
```
Comment (sushain97) [inline phenny]: critical/fatal (I can't remember which function it is)

File: phenny [hunk 8]
```diff
@@ -148,8 +149,8 @@ def config_names(config):
     if os.path.isdir(there): 
         return files(there)
 
-    print("Error: Couldn't find a config file!", file=sys.stderr)
-    print('What happened to ~/.phenny/default.py?', file=sys.stderr)
+    logger.error("Error: Couldn't find a config file!")
+    logger.error('What happened to ~/.phenny/default.py?')
```
Comment (sushain97) [inline phenny]: critical/fatal (I can't remember which function it is)

File: phenny [hunk 9]
```diff
@@ -196,9 +203,8 @@ def main(argv=None):
             module.password = None
 
         if module.host == 'irc.example.net': 
-            error = ('Error: you must edit the config file first!\n' + 
-                   "You're currently using %s" % module.filename)
-            print(error, file=sys.stderr)
+            logger.error('Error: you must edit the config file first!')
+            logger.error("You're currently using %s" % module.filename)
```
Comment (sushain97) [inline phenny]: critical/fatal (I can't remember which function it is)

File: phenny [hunk 10]
```diff
@@ -209,7 +215,7 @@ def main(argv=None):
     except ImportError: 
         try: from phenny import run
         except ImportError: 
-            print("Error: Couldn't find phenny to import", file=sys.stderr)
+            logger.error("Error: Couldn't find phenny to import")
```
Comment (sushain97) [inline phenny]: critical/fatal (I can't remember which function it is)

File: tools.py [hunk 11]
```diff
@@ -122,13 +126,15 @@ def new(phenny, input, old=old):
...
     paths = modified_paths + added_paths + removed_paths
+
     if comment is None:
         comment = "No commit message provided!"
     else:
         comment = re.sub("[\n\r]+", " ␍ ", comment.strip())
 
     basepath = os.path.commonprefix(paths)
-    print(basepath)
+    logger.debug(basepath)
```
Comment (sushain97) [inline tools.py]: can this just go?

**Comment by sushain97** [comment]
Also, build?

**Comment by Androbin** [comment]
I have addressed your comments, split the PR and fixed the build.
